packages/playwright-core/src/server/browserContext.ts

```diff
@@ -685,9 +689,12 @@ export function validateBrowserContextOptions(options: channels.BrowserNewContex
   if (options.proxy) {
     if (!browserOptions.proxy && browserOptions.isChromium && os.platform() === 'win32')
       throw new Error(`Browser needs to be launched with the global proxy. If all contexts override the proxy, global proxy will be never used and can be any string, for example "launch({ proxy: { server: 'http://per-context' } })"`);
+    if (!browserOptions.persistent && options.clientCertificates)
+      throw new Error('Cannot specify both proxy and clientCertificates');
```

Comment:
Let's also throw when `browserOptions.proxy` is set, because it won't be actually used for this context, which is unexpected.